### PR TITLE
Move viewers between tabs

### LINKED ISSUE #21: Move viewers between tabs
Add support for moving viewers (plots) between tabs

### PR DESCRIPTION
**Pull request by hbcarlos**
Fixes #21 

[Screencast from 04-26-2023 10:58:28 AM.webm](https://user-images.githubusercontent.com/26092748/234525436-187ce575-16c4-4ea7-a95b-da53f62c722d.webm)

Moving between tabs works now, but here is a minor issue. Since we are not updating the document when moving/resizing an item, the tab gets rerendered every time we carry an item to a different tab. Hence, each item comes back to its original position. I can fix it in a new PR.

### CODE REVIEW

File: glue_lab/glue_ydoc.py [hunk 1]
```diff
@@ -38,11 +49,14 @@ def set(self, value: str) -> None:
 
         tab_names: List[str] = contents.get("__main__", {}).get("tab_names", [])
         viewers = contents.get("__main__", {}).get("viewers", [])
-        tabs: Dict[str, List] = {}
+        tabs: Dict[str, Y.YMap] = {}
+        print("Tabs:", tab_names)
+        print("Viewers:", viewers)
```
Comment (martinRenou) [inline glue_lab/glue_ydoc.py:54]:
```suggestion
```

File: src/viewPanel/tabLayout.ts [hunk 2]
```diff
@@ -269,6 +269,14 @@ export class TabLayout extends Layout {
     }
   }
 
+  /**
+   * Remove a the items from gridstack.
```
Comment (martinRenou) [inline src/viewPanel/tabLayout.ts:273]:
```suggestion
   * Remove all items from gridstack.
```

**Review by martinRenou — COMMENTED** [review]
Looks great! Thanks

File: src/viewPanel/gridStackItem.ts [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 
 export class GridStackItem extends Panel {
   constructor(options: GridStackItem.IOptions) {
     super();
     this.removeClass('lm-Widget');
     this.removeClass('p-Widget');
     this.addClass('grid-stack-item');
+    this.addClass('glue-item');
 
     const { cellIdentity, cell, itemTitle = '', pos, size } = options;
```
Comment (trungleduc) [inline src/viewPanel/gridStackItem.ts:13]: Should we add this `cellIdentity` into an HTML attribute of this item? If so, in the `_evtContextMenu` handler, we can get the `cellIdentity` of the selected item immediately without doing the filtering logic in `_findItem`.

Comment (trungleduc) [reply src/viewPanel/gridStackItem.ts:13]: yes, if the target is a `GridStackItem` node and you set the `cellIdentity` attribute of this node before.

**Comment by trungleduc** [comment]
Thanks @hbcarlos

**Pull request merged**